### ISSUE TITLE
leading_exponent_vector segfaults when given zero polynomial

### ISSUE THREAD

**Issue by YueRen** [oscar-system/Singular.jl#544]
`leading_coefficient`, `leading_monomial`, `leading_term` work fine, but `leading_exponent_vector` crashes. Not sure what the correct output should be (or whether there is a correct output in the first place), but for now it would be better if the code gives an error instead of crashing:

```
julia> r,(x,y) = Singular.PolynomialRing(QQ, ["x","y"]);

julia> g = x-x;

julia> Singular.leading_coefficient(g)
0

julia> Singular.leading_monomial(g)
0

julia> Singular.leading_term(g)
0

julia> Singular.leading_exponent_vector(g)

signal (11): Segmentation fault
in expression starting at REPL[16]:1
p_GetExp at /workspace/destdir/include/singular/polys/monomials/p_polys.h:479 [inlined]
```

**Comment by tthsqe12** [comment]
@wbhart any idea why `leading_exponent_vector` is in Singular.jl but not part of the MPoly interface?

**Comment by tthsqe12** [comment]
@YueRen would you like this to give an error?

**Comment by wbhart** [comment]
Probably because leading anything applies more to Singular than to general stuff. But otherwise, no idea.

**Comment by YueRen** [comment]
I think giving an error is a fine choice for now. If a general consensus what the leading_exponent_vector of 0 is ever forms (which I doubt), we can always change it afterwards.

And I think leading_exponent_vector is in Singular because it depends strongly on Singular orderings.

**Comment by wbhart** [comment]
I think we had a bunch of functions lt, lm, lc which we added because that's what people wanted in Singular.jl. Then they all got renamed and maybe we added this one then?

**Comment by wdecker** [comment]
Two remarks:

1) Why speaking of leading_exponent_vector rather than leading_exponent (or to what else does leading_exponent_vector refer)?
2) In my eyes, the zero polynomial has leading_coefficient and leading_term zero by convention, but the leading_monomial and leading_exponent are not defined (and one should throw an error). However, there might be some reason to be consistent with Singular, @hannes14: What is the reason for Singular to behave differently from what I say above?

**Comment by wbhart** [comment]
leading_exponent_vector was just for consistency with exponent_vector

I'm happy for it to be changed.

**Comment by wdecker** [comment]
@wbhart Thanks for the explanation! Let me first try to see the whole picture before we start changing things.

**Comment by wbhart** [comment]
For further context, I think exponent was for just one exponent, e.g. the exponent of x, whereas exponent_vector was for the entire vector of exponents. So I think there actually is some consistency there.

**Comment by thofma** [comment]
Fixed by #545
